# HG changeset patch
# User tmodes
# Date 1710260877 -3600
#      Tue Mar 12 17:27:57 2024 +0100
# Node ID 4d081490b48aaff820cee7601b8624b37b652c06
# Parent  4b55f17c4e72d6c2f8b4930e3367ff52e1741b45
Fixed deprecated boost::filesystem::copy_option enum

diff -r 4b55f17c4e72 -r 4d081490b48a src/hugin_base/hugin_utils/filesystem.h
--- a/src/hugin_base/hugin_utils/filesystem.h	Tue Mar 12 17:27:29 2024 +0100
+++ b/src/hugin_base/hugin_utils/filesystem.h	Tue Mar 12 17:27:57 2024 +0100
@@ -64,6 +64,12 @@
     #endif
     #include <boost/filesystem.hpp>
     namespace fs = boost::filesystem;
-    #define OVERWRITE_EXISTING boost::filesystem::copy_option::overwrite_if_exists
+    #if BOOST_VERSION>=107400
+      // in Boost 1.74 and later filesystem::copy_option is deprecated
+      // use filesystem::copy_options instead
+      #define OVERWRITE_EXISTING boost::filesystem::copy_options::overwrite_existing
+    #else
+      #define OVERWRITE_EXISTING boost::filesystem::copy_option::overwrite_if_exists
+    #endif
 #endif
 #endif // _HUGIN_UTILS_FILESYSTEM_H
